src/main/java/org/folio/circulationbff/service/impl/CheckInServiceImpl.java

```diff
@@ -246,6 +246,19 @@ private void rebuildCheckInStaffSlipContext(CheckInResponse response, SearchInst
       .effectiveLocationLibrary(fetchLocationLibraryName(location.getLibraryId()))
       .effectiveLocationSpecific(location.getName());
 
+    if (item.getItemLevelCallNumber() != null) {
+      response.getStaffSlipContext().getItem()
+        .callNumber(item.getItemLevelCallNumber())
+        .callNumberPrefix(item.getItemLevelCallNumberPrefix())
+        .callNumberSuffix(item.getItemLevelCallNumberSuffix());
+    } else {
+      HoldingsRecord holdingsRecord = holdingsStorageClient.findHoldingsRecord(item.getHoldingsRecordId());
```

Comment:
Could it be that SearchInstance already has this info and we don't need to re-request it?